### PR TITLE
chore(node)!: Hide internal components

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/peer_tracker.rs [hunk 1]
```diff
@@ -207,30 +206,20 @@ impl PeerTracker {
...
-            if !peer_info.addrs.contains(addr) {
-                peer_info.addrs.push(addr.to_owned());
-            }
-        }
-
-        peer_info.state = PeerState::Identified;
-    }
-
     /// Returns true if peer is connected.
+    #[allow(dead_code)]
```
Comment (zvolin) [inline node/src/peer_tracker.rs:210]: why do we need this?

Comment (zvolin) [reply node/src/peer_tracker.rs:210]: I guess `pub(crate)` could do

File: node/src/p2p.rs [hunk 2]
```diff
@@ -240,13 +241,22 @@ impl P2p {
         let peer_tracker = Arc::new(PeerTracker::new(args.event_pub.clone()));
         let peer_tracker_info_watcher = peer_tracker.info_watcher();
 
-        let mut worker = Worker::new(args, cmd_rx, header_sub_tx, peer_tracker)?;
+        let cancellation_token = CancellationToken::new();
```
Comment (zvolin) [inline node/src/p2p.rs:244]: did you intend to include changes wrt shutdown on this pr?

Comment (oblique) [reply node/src/p2p.rs]: no. removed

File: node/src/peer_tracker.rs (hunk 1 above)
Comment (oblique) [reply node/src/peer_tracker.rs:210]: `is_connected` is not really used. even with `pub(crate)` we have a warning on the compilation, but I didn't want to remove the functionality.

File: node/src/block_ranges.rs [hunk 3]
```diff
@@ -7,10 +7,12 @@ use std::ops::{Add, RangeInclusive, Sub};
 use serde::Serialize;
 use smallvec::SmallVec;
 
-/// Type alias to `RangeInclusive<u64>`.
+/// Type alias to [`RangeInclusive<u64>`].
```
Comment (fl0rek) [inline node/src/block_ranges.rs:10]:
```suggestion
/// Type alias of [`RangeInclusive<u64>`].
```

File: node/src/daser.rs [hunk 4]
```diff
@@ -56,39 +56,39 @@ const SAMPLING_WINDOW: Duration = Duration::from_secs(30 * DAY);
 
 type Result<T, E = DaserError> = std::result::Result<T, E>;
 
-/// Representation of all the errors that can occur when interacting with the [`Daser`].
+/// Representation of all the errors that can occur from `Daser` component.
```
Comment (fl0rek) [inline node/src/daser.rs:59]:
```suggestion
/// Representation of all the errors that can occur in `Daser` component.
```
or 
```suggestion
/// Representation of all the errors that can be returned from `Daser` component.
```

File: node/src/node.rs [hunk 5]
```diff
@@ -18,36 +17,44 @@ use libp2p::identity::Keypair;
...
-use crate::syncer::{Syncer, SyncerArgs, SyncerError, SyncingInfo};
+use crate::syncer::{Syncer, SyncerArgs};
 
-type Result<T, E = NodeError> = std::result::Result<T, E>;
+pub use crate::daser::DaserError;
+pub use crate::p2p::{HeaderExError, P2pError};
+pub use crate::peer_tracker::PeerTrackerInfo;
+pub use crate::syncer::{SyncerError, SyncingInfo};
+
+/// Alias for a [`Result`] with the error type [`NodeError`]
```
Comment (fl0rek) [inline node/src/node.rs:36]:
```suggestion
/// Alias of [`Result`] with [`NodeError`] error type
```

File: node/src/node.rs [hunk 6]
```diff
@@ -1,8 +1,7 @@
-//! High-level integration of [`P2p`], [`Store`], [`Syncer`].
+//! Node that connects to Celestia's P2P network.
 //!
-//! [`P2p`]: crate::p2p::P2p
-//! [`Store`]: crate::store::Store
-//! [`Syncer`]: crate::syncer::Syncer
+//! Node will connect to Celestia's P2P network then will proceed
```
Comment (fl0rek) [inline node/src/node.rs:3]:
```suggestion
//! Upon creation, `Node` will try to connect to Celestia's P2P network and then proceed
```
slightly more specific about the timeline of events, wdyt?

File: node/src/p2p.rs [hunk 7]
```diff
@@ -86,7 +86,7 @@ const FRAUD_PROOF_HEAD_HEIGHT_THRESHOLD: u64 = 20;
 
 type Result<T, E = P2pError> = std::result::Result<T, E>;
 
-/// Representation of all the errors that can occur when interacting with [`P2p`].
+/// Representation of all the errors that can occur from `P2p` component.
```
Comment (fl0rek) [inline node/src/p2p.rs:89]: similar to daser,
```suggestion
/// Representation of all the errors that can occur in `P2p` component.
```
or
```suggestion
/// Representation of all the errors that can be returned from `P2p` component.
```

File: node/src/p2p/header_ex.rs [hunk 8]
```diff
@@ -65,7 +65,7 @@ pub(crate) struct HeaderExConfig<'a, S> {
     pub header_store: Arc<S>,
 }
 
-/// Representation of all the errors that can occur when interacting with the header-ex.
+/// Representation of all the errors that can occur from `HeaderEx` component.
```
Comment (fl0rek) [inline node/src/p2p/header_ex.rs:68]:
```suggestion
/// Representation of all the errors that can occur in `HeaderEx` component.
```
```suggestion
/// Representation of all the errors that can be returned from `HeaderEx` component.
```

File: node/src/syncer.rs [hunk 9]
```diff
@@ -38,22 +38,22 @@ type Result<T, E = SyncerError> = std::result::Result<T, E>;
 const TRY_INIT_BACKOFF_MAX_INTERVAL: Duration = Duration::from_secs(60);
 const SYNCING_WINDOW: Duration = Duration::from_secs(30 * 24 * 60 * 60); // 30 days
 
-/// Representation of all the errors that can occur when interacting with the [`Syncer`].
+/// Representation of all the errors that can occur from `Syncer` component.
```
Comment (fl0rek) [inline node/src/syncer.rs:41]:
```suggestion
/// Representation of all the errors that can occur in `Syncer` component.
```
```suggestion
/// Representation of all the errors that can be returned from `Syncer` component.
```

File: node/src/node.rs (hunk 6 above)
Comment (oblique) [reply node/src/node.rs]: looks good

**Review by fl0rek — APPROVED** [review]
nice work! 🙇

**Pull request merged**